README.rst

```diff
@@ -14,23 +14,103 @@ Lasagne
 
-* Simplicity: it should be easy to use and extend the library. Whenever a feature is added, the effect on both of these should be considered. Every added abstraction should be carefully scrutinized, to determine whether the added complexity is justified.
+Example
+-------
 
-* Small interfaces: as few classes and methods as possible. Try to rely on Theano's functionality and data types where possible, and follow Theano's conventions. Don't wrap things in classes if it is not strictly necessary. This should make it easier to both use the library and extend it (less cognitive overhead).
+.. code-block:: python
 
-* Don't get in the way: unused features should be invisible, the user should not have to take into account a feature that they do not use. It should be possible to use each component of the library in isolation from the others.
+  import lasagne
+  import theano
+  import theano.tensor as T
 
-* Transparency: don't try to hide Theano behind abstractions. Functions and methods should return Theano expressions and standard Python / numpy data types where possible.
+  # create Theano variables for input and target minibatch
+  input_var = T.tensor4('X')
+  target_var = T.ivector('y')
 
-* Focus: follow the Unix philosophy of "do one thing and do it well", with a strong focus on feed-forward neural networks.
+  # create a small convolutional neural network
+  from lasagne.nonlinearities import leaky_rectify, softmax
+  network = lasagne.layers.InputLayer((None, 3, 32, 32), input_var)
+  network = lasagne.layers.Conv2DLayer(network, 64, (3, 3),
+                                       nonlinearity=leaky_rectify)
+  network = lasagne.layers.Conv2DLayer(network, 32, (3, 3),
+                                       nonlinearity=leaky_rectify)
+  network = lasagne.layers.Pool2DLayer(network, (3, 3), stride=2, mode='max')
+  network = lasagne.layers.DenseLayer(lasagne.layers.dropout(network, 0.5),
+                                      128, nonlinearity=leaky_rectify,
+                                      W=lasagne.init.Orthogonal())
+  network = lasagne.layers.DenseLayer(lasagne.layers.dropout(network, 0.5),
+                                      10, nonlinearity=softmax)
+
+  # create loss function
+  prediction = lasagne.layers.get_output(network)
+  loss = lasagne.objectives.categorical_crossentropy(prediction, target_var)
+  loss = loss.mean() + 1e-4 * lasagne.regularization.regularize_network_params(
+          network, lasagne.regularization.l2)
+
+  # create parameter update expressions
+  params = lasagne.layers.get_all_params(network, trainable=True)
+  updates = lasagne.updates.nesterov_momentum(loss, params, learning_rate=0.01,
+                                              momentum=0.9)
+
+  # compile training function that updates parameters and returns training loss
+  train_fn = theano.function([input_var, target_var], loss, updates=updates)
+
+For a fully-functional example, see `examples/mnist.py <examples/mnist.py>`_,
```

Comment:
Yes, maybe... I mainly stopped because we're leaving the scope of Lasagne, and what's missing (reading data, training and testing) is included in the long example. The idea was to show how networks and update rules can be created, because that's all Lasagne can do for now, but of course that's only sufficient if you know Theano.

Are you thinking of something like:
```
# train network (assuming you've already got some training data as numpy arrays)
for epoch in range(100):
    loss = N = 0
    for input_batch, target_batch in training_data:
        loss += train_fn(input_batch, target_batch)
        N += 1
    print ("Epoch %d: Loss %g" % (epoch + 1, loss / N))

# use trained network for predictions
predict_fn = theano.function(
        [input_var],
        T.argmax(lasagne.layers.get_output(network, deterministic=True), axis=1))
print ("Predicted class for first ten test inputs: %r" % predict_fn(test_data[:10]))
```
Suggestions are highly welcome!